File: surf-data-api/src/main/java/dev/slne/surf/data/api/meta/SurfRedisRepository.java
package dev.slne.surf.data.api.meta;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;
import org.springframework.core.annotation.AliasFor;
import org.springframework.stereotype.Repository;

@Target(ElementType.TYPE)
@Retention(RetentionPolicy.RUNTIME)
@Repository
public @interface SurfRedisRepository {

  @AliasFor(annotation = Repository.class)
  String value() default "";

}

File: surf-cloud-api/surf-cloud-api-server/src/main/kotlin/dev/slne/surf/cloud/api/server/meta/SurfRedisRepository.kt
package dev.slne.surf.cloud.api.server.meta

import org.springframework.core.annotation.AliasFor
import org.springframework.stereotype.Repository

/**
 * Annotation for marking a class as a Redis repository in the Surf Cloud application.
 *
 * @property value The bean name of the repository.
 */
@Target(AnnotationTarget.CLASS)
@Retention(AnnotationRetention.RUNTIME)
@Repository
annotation class SurfRedisRepository(@get:AliasFor(annotation = Repository::class) val value: String = "")
